internal/handlers/common/filter.go

```diff
@@ -58,11 +59,33 @@ func filterDocumentPair(doc *types.Document, filterKey string, filterValue any)
 		if err != nil {
 			return false, nil // no error - the field is just not present
 		}
-		var ok bool
-		if doc, ok = docValue.(*types.Document); !ok {
-			return false, nil // no error - the field is just not present
+
+		switch docValue := docValue.(type) {
+		case *types.Document:
+			doc = docValue
+			filterKey = path.Suffix()
+		case *types.Array:
+			index, err := strconv.Atoi(path.Suffix())
+			if err != nil {
+				return false, nil
+			}
+
+			if docValue.Len() == 0 {
+				return false, nil
+			}
+
+			value, err := docValue.Get(index)
+			if err != nil {
+				return false, err
```

Comment:
Can you please add a test with an index value greater than array size to test this line?